src/clients/FtpGateway.cc

```diff
@@ -1008,6 +1008,11 @@ Ftp::Gateway::processReplyBody()
 
     entry->flush();
 
+    if (theSize >= 0 && data.payloadSeen >= theSize) {
+        markParsedVirginReplyAsWhole("whole virgin body");
+        completeForwarding();
```

Comment:
This fixes 'ABORTED' for ftp 'GET' transactions. We cannot add this to ftpReadTransferDone() similarly as to ftpWriteTransferDone() for 'PUT' transactions because for 'GET' transactions 226/250 response may come after Squid sends the whole downloaded file to the client (and client may have closed the connection before 226/250). Note that this does not happen for 'PUT' transactions, because ftpWriteTransferDone() initiates the response itself, i.e., the client receives the response only after Squid gets 226/250.